llpc/util/llpcElfWriter.cpp

```diff
@@ -735,6 +737,285 @@ template <class Elf> void ElfWriter<Elf>::updateMetaNote(Context *pContext, cons
+    while (psStartPos < relocCount) {
+      if (relocs[psStartPos].r_offset >= psIsaOffset)
+        break;
+      psStartPos++;
+    }
+  }
+
+  return psStartPos;
+}
+
+// =====================================================================================================================
+// Create a new empty section based on the input section
+//
+// @param [in] pName : Name of the new section
+// @param [in] pInputSection : the input section
+// @param [out] pNewSection : new section returned
+template <class Elf>
+void ElfWriter<Elf>::createNewSection(const char *pName, SectionBuffer *pInputSection, SectionBuffer *pNewSection) {
+
+  // Clear the new section
+  memset(pNewSection, 0, sizeof(SectionBuffer));
+
+  pNewSection->secHead = pInputSection->secHead;
+
+  // Add the section name to .strtab section
+  unsigned strTabSize = m_sections[m_strtabSecIdx].secHead.sh_size;
+  unsigned secNameSize = strlen(const_cast<char *>(pName)) + 1;
+
+  pNewSection->secHead.sh_name = strTabSize;
+  pNewSection->name = const_cast<char *>(pName);
+
+  pNewSection->secHead.sh_link = m_symSecIdx;
+  pNewSection->secHead.sh_size = 0;
+
+  auto strTabData = new uint8_t[strTabSize + secNameSize];
+  memcpy(strTabData, m_sections[m_strtabSecIdx].data, strTabSize);
+  memcpy(strTabData + strTabSize, pName, secNameSize);
+
+  delete[] m_sections[m_strtabSecIdx].data;
+
+  m_sections[m_strtabSecIdx].data = strTabData;
+  m_sections[m_strtabSecIdx].secHead.sh_size = strTabSize + secNameSize;
+}
+
+// =====================================================================================================================
+// Add the reloc related symbols and sections based on the input reloc
+//
+// @param reader : The ElfReader to the input fragment elf bin
+// @param inputReloc : The input reloc
+template <class Elf>
+void ElfWriter<Elf>::addRelocSymbols(const ElfReader<Elf> &reader, typename Elf::Reloc inputReloc) {
+  ElfSymbol relocSym = {};
+  reader.getSymbol(inputReloc.r_symbol, &relocSym);
+
+  // Create a section named origSectionName.cached
+  char *secName = new char[strlen(relocSym.secName) + strlen(CachedRodataSectionSuffix)];
+  strcpy(secName, relocSym.secName);
+  strcat(secName, CachedRodataSectionSuffix);
+
+  for (auto section : m_sections) {
+    if (strcmp(section.name, secName) == 0) {
+      return;
+    }
+  }
+
+  // If the section does not exist, create a new section and update the related symbols
+  SectionBuffer newSection = {};
+  ElfSectionBuffer<Elf64::SectionHeader> *inputRodataSection = nullptr;
+  std::vector<ElfSymbol> inputRodataSymbols;
+  auto inputRodataSecIndex = reader.GetSectionIndex(relocSym.secName);
+  reader.getSectionDataBySectionIndex(inputRodataSecIndex, &inputRodataSection);
+  reader.GetSymbolsBySectionIndex(inputRodataSecIndex, inputRodataSymbols);
+
+  createNewSection(secName, inputRodataSection, &newSection);
+
+  // Update the data from the input section
```

Comment:
nit: missing period at the end of the comment

Reply:
Addressed.